doc/dev/tutorial.md

```diff
@@ -0,0 +1,239 @@
+# Juju Developer Tutorial
+
+To get acquainted with the insides of Juju, you're going to add a brand-new
+feature. A quote-of-the-day service. With your addition, you will be able to set
+an author for today's quote and Juju will fetch one of their venerable uttering.
+Charms will then be able to fetch and use these words of high wisdom to increase
+their effectiveness as they see fit.
+
+**What you'll need:**
+- A workstation with `git` and an editor suitable for development in golang.
+
+**What you'll do:**
+- Add a new command to the Juju CLI to set the author of today's quote.
+- Add an API endpoint and facade method to the controller to save the new quote author.
+- Add a domain to the controller for storing the quote authors.
+- Add a worker to the controller which watches the state for changes, fetches quotes from new authors in state from
+  [Zenquotes API](https://docs.zenquotes.io/zenquotes-documentation/#call-author), and saves these quotes in state.
+- Add a hook tool for charms to get the quote of the day from the controller.
+- Write a basic charm to fetch the quote of the day, and display it in its status.
```

Comment:
Maybe comment out the lines that have not yet been implemented?

Reply:
Will prepend then with "(Still to write)" for now, imo its good to have them visible since I'm actively working on this as they should actually materialise in the near future (if that changes then happy to comment them out)